stdlib/public/core/Assert.swift

```diff
@@ -297,6 +297,22 @@ internal func _internalInvariant(
 #endif
 }
 
+// Only perform the invariant check on Swift 5.1 and later
+@_alwaysEmitIntoClient // Swift 5.1
+@_transparent
+internal func _internalInvariant_5_1(
+  _ condition: @autoclosure () -> Bool, _ message: StaticString = StaticString(),
+  file: StaticString = #file, line: UInt = #line
+) {
+#if INTERNAL_CHECKS_ENABLED
+  // FIXME: The below won't run the assert on 5.1 stdlib if testing on older
+  // OSes, which means that testing may not test the assertion. We need a real
+  // solution to this.
+  guard #available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *) else { return }
```

Comment:
Yeah, this doesn't look like the right check at all. You can't *ever* be sure that all code in a process is 5.1 code unless you just built it.

Reply:
This is only enabled in debug builds of the stdlib, though.

Granted, that doesn't rule out components linked from 5.0-era binaries, but it does make them less likely. It seems to me that the value of having the internal checks in place for the overwhelmingly likely case is worth the price of people running into false positives when doing esoteric things like *partially* recompiling their apps using a custom debug toolchain.

Reply:
That's probably okay for 5.1, but won't be okay for 5.2, where binary frameworks are a supported thing.